erpnext/accounts/doctype/financial_report_template/financial_report_validation.py

```
@@ -0,0 +1,592 @@
+					"pow": pow,
+					"ceil": lambda x: int(x) + (1 if x % 1 else 0),
+					"floor": lambda x: int(x),
+				}
+			)
+
+			result = frappe.safe_eval(formula, eval_globals=None, eval_locals=context)
+
+			if not isinstance(result, int | float):
+				return f"Formula must return a numeric value, got {type(result).__name__}"
+
+			return None
```

Ruff (0.13.3)

Consider moving this statement to an `else` block

(TRY300)